### PR TITLE
JDK-8316030: Update Libpng to 1.6.40

### PR DESCRIPTION
**Pull request by honkar-jdk**
Libpng updated from 1.6.39 to 1.6.40

Build and CI Testing passes on all platforms.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316030](https://bugs.openjdk.org/browse/JDK-8316030): Update Libpng to 1.6.40 (**Bug** - P3)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16223/head:pull/16223` \
`$ git checkout pull/16223`

Update a local copy of the PR: \
`$ git checkout pull/16223` \
`$ git pull https://git.openjdk.org/jdk.git pull/16223/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16223`

View PR using the GUI difftool: \
`$ git pr show -t 16223`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16223.diff">https://git.openjdk.org/jdk/pull/16223.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16223#issuecomment-1766895513)

### CODE REVIEW

**Review by prrace — APPROVED** [review]
A very small update

**Review by DamonGuy — APPROVED** [review]
Changes look good to me. I left one question for clarification for myself.

File: src/java.desktop/share/native/libsplashscreen/libpng/pngget.c [hunk 1]
```diff
@@ -49,7 +49,18 @@ png_get_valid(png_const_structrp png_ptr, png_const_inforp info_ptr,
...
    if (png_ptr != NULL && info_ptr != NULL)
+   {
+#ifdef PNG_READ_tRNS_SUPPORTED
+      /* png_handle_PLTE() may have canceled a valid tRNS chunk but left the
+       * 'valid' flag for the detection of duplicate chunks. Do not report a
+       * valid tRNS chunk in this case.
+       */
+      if (flag == PNG_INFO_tRNS && png_ptr->num_trans == 0)
+         return(0);
+#endif
```
[the remark this answers is not in the document]
Comment (honkar-jdk) [reply src/java.desktop/share/native/libsplashscreen/libpng/pngget.c:60]: This is an upstream issue. 
Details about the issue and fix is available on libpng source repo.

**Pull request closed without merging**